src/corelibs/analogio/test_analogio_pwm.cpp

```diff
@@ -240,6 +240,25 @@ TEST_IFX(analogio_pwm, test_analog_write_pwm_0_percentage_dutycycle)
     TEST_ASSERT_EQUAL_MESSAGE(LOW, digitalRead(PWM_PIN_FEEDBACK), "PWM output should be LOW when 0 percentage duty cycle is set");
 }
 
+/**
+ * @brief Verify SetAnalog frequency
+ */
+TEST_IFX(analogio_pwm, test_analog_set_fz)
+{
+    analogWriteResolution(16); // Set resolution to 16 bits
```

Comment:
Based on the comment of the func implementation, shall we have a test where we set the frequency before calling analogWrite()?